File: Otros/Robot_model/Dir_model.py
import sys
import numpy as np
import math

T = np.array([0, 0, 0])

for i in range(1, len(sys.argv)):
    T[i-1] = float(sys.argv[i])

P = np.array([[0], [0], [0], [1]])
a = np.array([0, 10, 10])

A01 = np.array([[math.cos(T[0]), 0, math.sin(T[0]), 0], 
              [math.sin(T[0]), 0, -math.cos(T[0]), 0 ],
              [0, 1, 0, 0],
              [0, 0, 0, 1]])

A12 = np.array([[math.cos(T[1]), -math.sin(T[1]), 0, a[1]*math.cos(T[1])],
                [math.sin(T[1]), math.cos(T[1]), 0, a[1]*math.sin(T[1])],
                [0, 0, 1, 0],
                [0, 0, 0, 1]])

A23 = np.array([[math.cos(T[2]), -math.sin(T[2]), 0, a[2]*math.cos(T[2])],
                [math.sin(T[2]), math.cos(T[2]), 0, a[2]*math.sin(T[2])],
                [0, 0, 1, 0],
                [0, 0, 0, 1]])

A03 = np.matmul(np.matmul(A01,A12),A23)

EF = np.matmul(A03,P)
print(EF)

File: Otros/Robot_model/Dir_model.m
function [EF]=Dir_model(T)
%Dir_model: Modelo cinemático directo brazo antropomorfico 3DOF
%Input
%   T: vector con ángulos de articulación (radianes)
%Output
%   EF: Posición del efector final desde hombro

 P = [0; 0; 0; 1];

 a = [0 10 10];

 A01 = [cos(T(1)) 0 sin(T(1)) 0;
        sin(T(1)) 0 -cos(T(1)) 0;
        0 1 0 0;
        0 0 0 1];

  A12 = [cos(T(2)) -sin(T(2)) 0 a(2)*cos(T(2));
         sin(T(2)) cos(T(2)) 0 a(2)*sin(T(2));
         0 0 1 0;
         0 0 0 1];
  A23 = [cos(T(3)) -sin(T(3)) 0 a(3)*cos(T(3));
         sin(T(3)) cos(T(3)) 0 a(3)*sin(T(3));
         0 0 1 0;
         0 0 0 1];
     
  A03 = A01*A12*A23;
  
  EF = A03*P;
end